### PR TITLE
117 results page consolidate 3 views into one

### PR DESCRIPTION
**Pull request by Dhruv-Copods**
Made following changes,

- Refactored result table 
- Removed extra code

### CODE REVIEW

File: app/components/results/CandidatesList.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            ).toFixed(2)
+          ),
+        })
+      }
+    }
+    return resultData
+  }
+
+  return (
+    <div className="bg-[#F9FAFB] pb-4">
```
Comment (arpita-20) [inline app/components/results/CandidatesList.tsx:34]: remove hardcoded value ,add gray-50

File: app/models/result.server.ts [hunk 2]
```diff
@@ -81,6 +81,42 @@ export async function getResultsOfCandidatesByTestId({
   })
 }
 
+
```
Comment (arpita-20) [inline app/models/result.server.ts:84]: remove extra spaces

File: app/routes/results/groupByTests/$testId.tsx [hunk 3]
```diff
@@ -5,23 +5,50 @@ import type { LoaderFunction } from '@remix-run/server-runtime'
 import { json } from '@remix-run/node'
 
 import invariant from 'tiny-invariant'
-import { getPendingExamCandidateByTestId } from '~/models/result.server'
+import {
+  getAllCandidatesOfTest,
+  // getPendingExamCandidateByTestId,
```
Comment (arpita-20) [inline app/routes/results/groupByTests/$testId.tsx:10]: remove commented code

File: app/routes/results/groupByTests/$testId.tsx [hunk 4]
```diff
@@ -5,23 +5,50 @@ import type { LoaderFunction } from '@remix-run/server-runtime'
...
+  // getResultsOfCandidatesByTestId,
+} from '~/models/result.server'
 
-type LoaderData = {
-  testPreview: Awaited<ReturnType<typeof getPendingExamCandidateByTestId>>
-  params: any
-}
+// type LoaderData = {
```
Comment (arpita-20) [inline app/routes/results/groupByTests/$testId.tsx:14]: remove commented code

File: app/routes/results/groupByTests/$testId.tsx [hunk 5]
```diff
@@ -5,23 +5,50 @@ import type { LoaderFunction } from '@remix-run/server-runtime'
...
+//   testPreview: Awaited<ReturnType<typeof getPendingExamCandidateByTestId>>
+//   params: any
+// }
 
 export const loader: LoaderFunction = async ({ request, params }) => {
+  // invariant(params.testId, 'resultId not found')
+  // const testPreview = await getPendingExamCandidateByTestId({
+  //   id: params.testId,
+  // })
```
Comment (arpita-20) [inline app/routes/results/groupByTests/$testId.tsx:23]: remove commented code

File: app/routes/results/groupByTests/$testId.tsx [hunk 6]
```diff
@@ -5,23 +5,50 @@ import type { LoaderFunction } from '@remix-run/server-runtime'
...
   })
-  if (!testPreview) {
+  if (!candidatesOfTest) {
     throw new Response('Not Found', { status: 404 })
   }
 
-  return json<LoaderData>({ testPreview, params })
+  return json({
+    // testPreview,
+    // attendedCandidateForTest,
```
Comment (arpita-20) [inline app/routes/results/groupByTests/$testId.tsx:48]: remove all commented code

File: app/components/results/ResultDetails.tsx [hunk 7]
```diff
@@ -38,7 +38,7 @@ const ResultDetailsComponent = () => {
       <header>
         <div className="flex gap-2">
           <NavLink
-            to={`/results/groupByTests/${params?.testId}/completed`}
+            to={`/results/groupByTests/${params?.testId}`}
             className="flex items-center gap-4 "
```
Comment (arpita-20) [inline app/components/results/ResultDetails.tsx:42]: remove extra space

File: app/components/results/CandidateListOfTest.tsx [hunk 8]
```diff
@@ -1,28 +1,11 @@
...
-    {
-      id: 2,
-      title: 'Completed',
-      route: 'completed',
-    },
-  ]
-  const { testPreview, params } = useLoaderData()
+  const { candidatesOfTest } = useLoaderData()
   return (
     <div id="test-details" className=" h-full ">
```
Comment (yash6102000) [inline app/components/results/CandidateListOfTest.tsx:10]: remove extra space

File: app/components/results/AttendedCandidateListItem.tsx [hunk 9]
```diff
@@ -37,15 +39,17 @@ const AttendedCandidateListItem = ({
           <span className="text-base text-gray-700">{invitedBy}</span>
         </div>
         <div className="col-span-1 truncate">
-          <span className=" text-base text-gray-700">{result}%</span>
+          <span className=" text-base text-gray-700">
```
Comment (yash6102000) [inline app/components/results/AttendedCandidateListItem.tsx:42]: remove extra space

File: app/components/results/CandidatesList.tsx (hunk 1 above)
Comment (yash6102000) [inline app/components/results/CandidatesList.tsx:34]: remove hard coded style

File: app/components/results/CandidatesList.tsx [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+      <div className="bg-tableHeader rounded-lg border border-solid border-gray-200 shadow-base">
```
Comment (yash6102000) [inline app/components/results/CandidatesList.tsx:35]: remove bg-tableheader

**Review by anuragP04 — APPROVED** [review]

**Pull request merged**